### PR TITLE
Added some javadocs

### PR DESCRIPTION
**Pull request by Adrijaned**
Added javadocs to `CurvedSegment`, `LinearSegment` and `SegmentMapping`

### CODE REVIEW

**Comment by Adrijaned** [comment]
Aldo added javadocs to `PathFollowerSystem` and `SegmentSystem`. Added to the previous commit

File: src/main/java/org/terasology/segmentedpaths/controllers/PathFollowerSystem.java [hunk 1]
```diff
@@ -42,10 +42,24 @@
...
+     * @return Vector representation of the tangent
+     */
     public Vector3f vehicleTangent(EntityRef vehicleEntity) {
         return vehicleTangent(vehicleEntity, 0, null);
     }
 
+    /**
+     * Returns a tangent to the path at the point the vehicle will be after delta distance.
+     *
+     * @param vehicleEntity Vehicle entity to which we want a tangent
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/segmentedpaths/controllers/PathFollowerSystem.java:58]: If you're aligning all the parameter descriptions, do it for _the whole file_, align the return descriptions too. Looks far more consistent. As far as I know, most of our Javadocs don't align param descriptions, so you might just want to scrap it.

File: src/main/java/org/terasology/segmentedpaths/controllers/PathFollowerSystem.java [hunk 2]
```diff
@@ -125,6 +170,13 @@ public boolean isVehicleValid(EntityRef vehicleEntity) {
         return true;
     }
 
+    /**
+     * Moves the vehicle for delta distance along the path
+     * @param vehicleEntity Vehicle to be moved
+     * @param delta Distance the vehicle is to be moved by
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/segmentedpaths/controllers/PathFollowerSystem.java]: If you're aligning param descriptions in other places, do it here too. Stay consistent

File: src/main/java/org/terasology/segmentedpaths/controllers/PathFollowerSystem.java (hunk 1 above)
Comment (mjuvekar7) [reply src/main/java/org/terasology/segmentedpaths/controllers/PathFollowerSystem.java:58]: I would recommend just not aligning at all

File: src/main/java/org/terasology/segmentedpaths/controllers/PathFollowerSystem.java [hunk 3]
```diff
@@ -111,7 +152,11 @@ public Vector3f vehicleNormal(EntityRef vehicleEntity, float delta, SegmentMappi
         return null;
     }
 
-
+    /**
+     * Returns whether the vehicle is valid for working with.
+     * @param vehicleEntity Vehicle that we want to check
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/segmentedpaths/controllers/PathFollowerSystem.java]: Blank line before this

**Pull request merged**